src/components/smartSearch/EditTargetDialog.tsx

```diff
@@ -0,0 +1,128 @@
+import { FormattedMessage as Msg } from 'react-intl';
+import { useState } from 'react';
+import { Box, Button, ButtonBase, Card, CardContent, Dialog, DialogContent, Typography } from '@material-ui/core';
+
+import All from './filters/All';
+import Filter from './Filter';
+import { isFilterWithId } from './utils';
+import MostActive from './filters/MostActive';
+import patchTaskTarget from 'fetching/tasks/patchTaskTarget';
+import { useRouter } from 'next/router';
+import { ZetkinSmartSearchFilter } from 'types/zetkin';
+import { FILTER_TYPE, ZetkinSmartSearchFilterWithId } from 'types/smartSearch';
+import { useMutation, useQueryClient } from 'react-query';
+
+interface EditTargetDialogProps {
+    filterSpec: ZetkinSmartSearchFilter[];
+    onDialogClose: () => void;
+    open: boolean;
+}
+
+const EditTargetDialog = ({ onDialogClose, open, filterSpec }: EditTargetDialogProps) : JSX.Element => {
```

Comment:
Not necessary right now, but let's help each other remember to rename this to something more general purpose for smart searches, like `EditSmartSearchDialog`, `EditQueryDialog` or something like that. 😊

Reply:
I think it's good to change it now personally! Edit smart search dialog is good I reckon. One thing is that this only edits the smart search filters for a task! So maybe `TaskAssigneesSmartSearchDialog`